dm/loader/lightning.go

```diff
@@ -320,13 +321,13 @@ func GetLightningConfig(globalCfg *lcfg.GlobalConfig, subtaskCfg *config.SubTask
 	cfg.Routes = subtaskCfg.RouteRules
 
 	if subtaskCfg.ExtStorage != nil {
-		// NOTE: if we use bucket as dumper storage, write lightning checkpoint to downstream DB to avoid bucket ratelimit
-		// it's the default path for cloud dm.
+		// NOTE: If we use bucket as dumper storage, write lightning checkpoint to downstream DB to avoid bucket ratelimit
 		// since we will use check Checkpoint in 'ignoreCheckpointError', MAKE SURE we have assigned the Checkpoint config properly here
 		if err := cfg.Security.BuildTLSConfig(); err != nil {
 			return nil, err
 		}
-		cfg.Checkpoint.Schema = "tidb_lightning_checkpoint"
+		// To enable the loader worker failover, we need to use jobID+sourceID to isolate the checkpoint schema
+		cfg.Checkpoint.Schema = cputil.LightningCheckpointSchema(workerName, subtaskCfg.SourceID)
```

Comment:
Should be task name+source id.
Worker is dynamically bound to source. If the dump files are stored in s3, another worker can continue the task from previous worker's checkpoint

Reply:
workerName is assigned to the  `JobID` outside as the comment says.

Reply:
not sure if lightning escape the database name.